solver_spec.rb

```diff
@@ -0,0 +1,27 @@
+require_relative 'solver'
+
+describe Solver do
+  context 'it should solve problems correctly with methods' do
+    solver = Solver.new
+    string1 = 'hello'
+    string2 = 'adam'
+    string3 = 'tochukwu'
+
+    it 'get factorial of number' do
+      expect(solver.factorial(5)).to eq 120
+      expect(solver.factorial(1)).to eq 1
+      expect { solver.factorial(-1) }.to raise_error('Number is negative')
```

Comment:
- [ ] Your tests for the factorial method should contain a special case scenario for when **`0`** is the input - the result should be **`1`**. Kindly add it.

Reply:
I have added the test case, thank you for the review.